Content.Shared/Containers/ItemSlot/ItemSlotsSystem.cs

```diff
@@ -336,10 +336,16 @@ public bool TryEject(EntityUid uid, ItemSlot slot, EntityUid? user, [NotNullWhen
         {
             item = null;
 
+            /// This handles logic with the slot itself
             if (!CanEject(slot))
                 return false;
 
             item = slot.Item;
+
+            /// This handles user logic
+            if (user != null && item != null && !_actionBlockerSystem.CanPickup(user.Value, item.Value))
+                return false;
```

Comment:
@ElectroJr idk if this makes sense

Reply:
Didn't break any other item slot stuff I tested